src/main/java/net/neoforged/neoforge/common/extensions/ContainerExtension.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.extensions;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.VanillaContainerWrapper;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Extension methods for {@link Container}. These methods are used by {@link VanillaContainerWrapper},
+ * to allow containers to be integrated in a transaction.
+ */
+public interface ContainerExtension {
+    private Container self() {
+        return (Container) this;
+    }
+
+    /**
+     * An extension of {@link Container#setItem(int, ItemStack)} that allows disabling committed actions.
+     *
+     * <p>If {@code insideTransaction} is {@code false}, changes (e.g. calling {@code setChanged} or making changes to the world) should be deferred until after the insideTransaction.
```

Comment:
This is backwards. If `insideTransaction` is `true`, we want to defer these changes.